recipes/ogre/1.x/conanfile.py

```diff
@@ -0,0 +1,320 @@
+        "ogre_build_rendersystem_gl3plus": True,
+        "ogre_build_samples": False,
+        "ogre_build_tests": False,
+        "ogre_build_tools": True,
+        "ogre_config_enable_quad_buffer_stereo": False,
+        "ogre_config_filesystem_unicode": True,
+        "ogre_config_threads": 3,
+        "ogre_config_thread_provider": "std",
+        "ogre_config_enable_freeimage": True,
+        "ogre_install_pdb": False,
+        "ogre_install_samples": False,
+        "ogre_install_tools": True,
+        "ogre_install_vsprops": False,
+        "ogre_resourcemanager_strict": 2,
+        "ogre_static": True,
+        "ogre_set_double": False,
+        "ogre_glsupport_use_egl": True
+    }
+    exports_sources = "CMakeLists.txt", "patches/**"
+    short_paths = True
+
+    def requirements(self):
+        self.requires("cppunit/1.15.1")
+        self.requires("freeimage/3.18.0")
+        self.requires("boost/1.75.0")
+        self.requires("freetype/2.11.1")
+        self.requires("openexr/2.5.7")
+        self.requires("poco/1.11.2")
+        self.requires("tbb/2020.3")
+        self.requires("zlib/1.2.12")
+        self.requires("zziplib/0.13.71")
+        self.requires("openssl/1.1.1o", override=True)
+        self.requires("xorg/system")
+        self.requires("glu/system")
+        self.requires("sdl/2.0.20")
+        if self.options.ogre_glsupport_use_egl and self.settings.os in ["Linux", "FreeBSD"]:
+            self.requires("egl/system")
+        else:
+            self.requires("libglvnd/1.4.0")
+        
+
+    def validate(self):
+        """
+         OGRE 1.x is very old and will not work with latest gcc, clang and msvc compielrs
+         TODO: determine incompatible msvc compilers
+        """
+        if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) >= 9:
+            raise ConanInvalidConfiguration("OGRE 1.x not supported with gcc version greater than 9")
+        if self.settings.compiler == "clang" and tools.Version(self.settings.compiler.version) >= 9:
+            raise ConanInvalidConfiguration("OGRE 1.x not supported with clang version greater than 9")
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    @functools.lru_cache(1)
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["OGRE_STATIC"] = not self.options.shared
+        cmake.definitions["OGRE_CONFIG_DOUBLE"] = self.options.ogre_set_double
+        cmake.definitions["OGRE_CONFIG_NODE_INHERIT_TRANSFORM"] = False
+        cmake.definitions["OGRE_GLSUPPORT_USE_EGL"] = self.options.ogre_glsupport_use_egl
+        if not tools.valid_min_cppstd(self, 11):
+            cmake.definitions["CMAKE_CXX_STANDARD"] = 11 # for OpenEXR
```

Comment:
```suggestion
        if self.settings.compiler.get_safe("cppstd"):
            tools.check_min_cppstd(self, 11)
        else:
            # INFO: OpenEXR requires C++11
            cmake.definitions["CMAKE_CXX_STANDARD"] = 11
```
Need to guarantee Conan settings with compiler flags, otherwise, it will be a mess.

Reply:
what is the difference in the original implementation and your suggestion ?

Reply:
IF cppstd is declared, it will raise an error if it's less then 11. Also, Conan will inject the configured cppstd version to CMAKE_CXX_STANDARD, as your cmake wrapper runs `conan_basic_setup`. The `cppstd` is None by default because it was a feature in Conan after 1.0, so Conan could not break a trivial behavior.